write/write.c

```diff
@@ -0,0 +1,200 @@
+/*
+ * write.c - write to another user
+ *
+ * Copyright (C) 2024: Luiz Antônio Rangel (takusuman)
+ *
+ * SPDX-Licence-Identifier: Zlib
+ *
+ *	from Unix 32V /usr/src/cmd/write.c	
+ *	Novemberr 6th, 1978.
+ *
+ * Copyright(C) Caldera International Inc. 2001-2002. All rights reserved.
+ *
+ * SPDX-Licence-Identifier: Caldera
+ */
+
+#include <fcntl.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <sys/wait.h>
+#include <signal.h>
+#include <strings.h>
+#include <string.h>
+#include <unistd.h>
+#include <utmpx.h>
+
+char	*strcat();
+char	*strcpy();
+struct	utmpx *ubuf;
+int	signum[] = {SIGHUP, SIGINT, SIGQUIT, 0};
+char	me[10]	= "???";
+char	*him;
+char	*mytty;
+char	histty[32];
+char	*histtya;
+char	*ttyname();
+char	*rindex();
+int	logcnt;
+FILE	*tf;
+
+void	main(int argc, char *argv[]);
+int	eof(void);
+int	timout(void);
+int	ex(char* bp);
+void	sigs(int (*sig)());
+void	usage(void);
+
+void main(int argc, char *argv[]) {
+	struct stat stbuf;
+	register int i;
+	int c1, c2, tfd;
+
+	if (argc < 2) {
+		usage();
+	}
+	him = argv[1];
+	if (argc > 2) {
+		histtya = argv[2];
+	}
+	
+	mytty = ttyname(2);
+	if (mytty == NULL) {
+		printf("Can't find your tty\n");
+		exit(1);
+	}
+	mytty = rindex(mytty, '/') + 1;
+	if (histtya) {
+		strcpy(histty, "/dev/");
+		strcat(histty, histtya);
+	}
+	while ((ubuf = getutxent()) != NULL) { 
+		if (strcmp(ubuf[0].ut_line, mytty)==0) {
+			for(i=0; i<8; i++) {
+				c1 = ubuf->ut_user[i];
+				if(c1 == ' ')
+					c1 = 0;
+				me[i] = c1;
+				if(c1 == 0)
+					break;
+			}
+		}
+		if(him[0] != '-' || him[1] != 0)
+		for(i=0; i<8; i++) {
```

Comment:
And that is a manual string comparison, neat!
Didn't UNIX had a strcmp() at the time?

Reply:
I will replace it with strncmp() and use the same limits that there was at the wall.c code.